### PR TITLE
Refs #1184 -- Reformatted code with Black.

### PR DESCRIPTION
**Pull request by CuriousLearner**
This formats the code with black and also ensures that flake8 doesn't conflict with its changes.

### CODE REVIEW

**Comment by CuriousLearner** [comment]
In a subsequent PR, I'll add black in the pre-commit hooks and also ensure that this commit can be ignored in git-blame.

**Comment by pauloxnet** [comment]
Have read to comments form @carltongibson on this issue ?
https://github.com/django/djangoproject.com/issues/1184#issuecomment-1263431945

Please in any case wait for reviews before merge this PR. There's a lot of changed file so it can be take long time to review.

All contribute in review this PR are welcome.

**Comment by CuriousLearner** [comment]
> Have read to comments form @carltongibson on this issue ? [#1184 (comment)](https://github.com/django/djangoproject.com/issues/1184#issuecomment-1263431945)
> 

Yes, I did :) The commit on this PR is from dj-bot as Carlton mentioned. I'll [follow up with a PR](https://github.com/django/djangoproject.com/pull/1224#issuecomment-1264409027) as mentioned to mark this commit to be ignored in git-blame rev. Let me know if there is anything else.

> Please in any case wait for reviews before merge this PR. There's a lot of changed file so it can be take long time to review.
> 

Of course :) 

> All contribute in review this PR are welcome.

**Review by camilonova — APPROVED** [review]
LGTM

**Comment by CuriousLearner** [comment]
Sure, I'll have a look later in the evening and get this done.

Thanks!

**Comment by CuriousLearner** [comment]
Hi @carltongibson 👋🏼  !

I tried to incorporate your review. The default line-length is now standardized to 88 chars. I had to do some manual changes in the actual `format black` commit. I've also added separate commit(s) for adding black to github action and adjusting the config. Please let me know if anything else is needed here.

The actual black commit(s) lint action failed since the config was later adjusted in the subsequent commit, but please let me know if I should adjust the config in format black commit too to make everything passed (I'm not sure what is the preference here).

Many thanks!

cc @pauloxnet

File: requirements/dev.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
 -r common.txt
 pre-commit~=2.20.0
-transifex-client==0.14.4
+transifex-client==0.12.5
```
Comment (pauloxnet) [inline requirements/dev.txt:3]: This change seem unrelated to this PR
Can you revert please ?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -12,6 +12,10 @@ repos:
         args: ["--ignore-case", "--unique"]
       - id: fix-encoding-pragma
         args: ["--remove"]
+  - repo: https://github.com/psf/black
+    rev: 22.1.0
```
Comment (pauloxnet) [inline .pre-commit-config.yaml:16]: Can you upgrade to the latest version ?

**Comment by pauloxnet** [comment]
@CuriousLearner can you rebase this PR and read the comments above ?

**Comment by carltongibson** [comment]
@CuriousLearner Sorry for the slow follow-up. If you can rebase and address the comments I will take a proper look next week. Super effort 🎁

**Review by carltongibson — COMMENTED** [review]
OK, I think this looks good. @CuriousLearner — have you got the bandwidth to update? (If not I can do it!) I think we should get it in then. 

Ref the pre-commit config, @adamchainz had me use the bot/GitHub app on the Daphne repo, and that works quite well… — I was going to extend it to the other Channels repos, if that's a measure. If @django/djangoproject-com-maintainters think so, I can add maybe it here too? (Permissions are per-repo…)

**Comment by CuriousLearner** [comment]
Hey @carltongibson 

Sorry for the late reply. I had to wrap-up a project this week at work. I'll get this done tomorrow morning.

Cheers!

**Comment by CuriousLearner** [comment]
Hi @carltongibson 

I've rebased the PR and address @pauloxnet 's review. Please have a look now.

**Comment by pauloxnet** [comment]
> Ref the pre-commit config, @adamchainz had me use the bot/GitHub app on the Daphne repo, and that works quite well… — I was going to extend it to the other Channels repos, if that's a measure. If @django/djangoproject-com-maintainters think so, I can add maybe it here too? (Permissions are per-repo…)

I agree, please do it @carltongibson 🙏🏻

**Review by pauloxnet — APPROVED** [review]

**Comment by pauloxnet** [comment]
> I've rebased the PR and address @pauloxnet 's review. Please have a look now.

Hi @CuriousLearner and thanks for your hard work on this PR.

I've just reviewed again and approved, but let wait for @carltongibson review and edits.

**Comment by adamchainz** [comment]
After setting up pre-commit.ci, we can drop the tox jobs to run black/flake8/isort, and related github actions steps.

**Comment by carltongibson** [comment]
> I agree, please do it @carltongibson 🙏🏻

OK, I've just added the pre-commit app to this repo. (The was a 400 error in the process, but logging in there, it says it worked... 😬)

**Comment by carltongibson** [comment]
@CuriousLearner Just FYI, since I saw your reaction I'm rebasing this (and will experiment with a "bad commit") now.

File: setup.cfg [hunk 3]
```diff
@@ -1,12 +1,15 @@
 [flake8]
-exclude = djangodocs,.tox
-ignore = F405,W504
-max-line-length = 160
+exclude = djangodocs,.tox,*/migrations/*
+ignore = F405,W504,W503
+max-line-length = 88
+per-file-ignores =
+    docs/tests.py: E501
```
Comment (carltongibson) [inline setup.cfg:6]: This is for the docs fixtures in this file, which have a longer length but don't seem worth manually reformatting. 

See https://results.pre-commit.ci/run/github/690903/1669801921.cpckKRxoTl-Z01SxoOmUxQ

**Review by carltongibson — APPROVED** [review]
This looks good to me. Thanks for your work @CuriousLearner 🏅

I've rebased and adjusted slightly: 

* Added a line-length ignore for the docs test file with the long fixtures. 
* Dropped the GHA linter step as per Adam's comment. (I kept the tox jobs, since they're handy for me at least to run locally.) 

For me this is ready to be (rebase and) merged as three commits. @pauloxnet given that I edited can I just ask you to glance and do that? Thanks!

**Comment by pauloxnet** [comment]
> For me this is ready to be (rebase and) merged as three commits. @pauloxnet given that I edited can I just ask you to glance and do that? Thanks!

@carltongibson as you asked, after reviewing I rebased and merged using the Github action for that. I hope that's exactly what you asked me to do. :-)

Thanks again to @CuriousLearner for his works here.

**Comment by carltongibson** [comment]
@pauloxnet perfect thanks! 

@CuriousLearner if you wanted to follow-up adding the Black commit to a `.git-blame-ignore-revs` that would 🍒

**Comment by CuriousLearner** [comment]
Thanks for the review and getting this in @pauloxnet & @carltongibson 

@carltongibson I've added the subsequent PR here: https://github.com/django/djangoproject.com/pull/1280

Thanks 🙌🏼
